diff --git a/CMakeLists.txt b/CMakeLists.txt
index fbebe99e7..c691e10f3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -289,8 +289,8 @@ endif()
 if(QT_STATIC)
    if(WIN32)
       if(DESIRED_QT_VERSION MATCHES 4)
-         # QtCore needs WSAAsyncSelect from Ws2_32.lib
-         set(QT_QTCORE_LIBRARY ${QT_QTCORE_LIBRARY} Ws2_32.lib)
+         # QtCore needs WSAAsyncSelect from ws2_32.lib
+         set(QT_QTCORE_LIBRARY ${QT_QTCORE_LIBRARY} ws2_32.lib)
          message("QT_QTCORE_LIBRARY: ${QT_QTCORE_LIBRARY}")
       endif()
    endif()
@@ -299,9 +299,13 @@ endif()
 
 set(BOOST_COMPONENTS system filesystem program_options iostreams)
 if(WIN32)
-    set(BOOST_COMPONENTS ${BOOST_COMPONENTS} locale zlib)
+    set(BOOST_COMPONENTS ${BOOST_COMPONENTS} locale)
 endif(WIN32)
 
+if (WIN32 AND NOT MINGW)
+    set(BOOST_COMPONENTS ${BOOST_COMPONENTS} libz)
+endif()
+
 IF(BOOST_STATIC)
     set(Boost_USE_STATIC_LIBS   ON)
 endif()
diff --git a/apps/openmw/CMakeLists.txt b/apps/openmw/CMakeLists.txt
index 554a3ac86..c44028f07 100644
--- a/apps/openmw/CMakeLists.txt
+++ b/apps/openmw/CMakeLists.txt
@@ -119,6 +119,7 @@ include_directories(
 )
 
 target_link_libraries(openmw
+    ${OPENMW_CXX_FLAGS}
     ${OSG_LIBRARIES}
     ${OPENTHREADS_LIBRARIES}
     ${OSGPARTICLE_LIBRARIES}
@@ -141,7 +142,7 @@ target_link_libraries(openmw
     components
 )
 
-if (ANDROID)
+if (ANDROID OR OPENMW_STATIC)
     set (OSG_PLUGINS
         -Wl,--whole-archive
     )
@@ -154,15 +155,20 @@ if (ANDROID)
     )
 
     target_link_libraries(openmw
-        EGL
-        android
-        log
         dl
         z
         ${OPENSCENEGRAPH_LIBRARIES}
         freetype
         jpeg
-    	png
+        png
+    )
+endif (ANDROID OR OPENMW_STATIC)
+
+if (ANDROID)
+    target_link_libraries(openmw
+        EGL
+        android
+        log
     )
 endif (ANDROID)
 
diff --git a/cmake/FindOSGPlugins.cmake b/cmake/FindOSGPlugins.cmake
index c210466c0..16b591529 100644
--- a/cmake/FindOSGPlugins.cmake
+++ b/cmake/FindOSGPlugins.cmake
@@ -29,9 +29,7 @@ foreach(_library ${OSGPlugins_FIND_COMPONENTS})
 
     set(${_library_uc}_DIR ${OSGPlugins_LIB_DIR}) # to help function osg_find_library
     set(_saved_lib_prefix ${CMAKE_FIND_LIBRARY_PREFIXES}) # save CMAKE_FIND_LIBRARY_PREFIXES
-    set(CMAKE_FIND_LIBRARY_PREFIXES "") # search libraries with no prefix
     osg_find_library(${_library_uc} ${_library}) # find it into ${_library_uc}_LIBRARIES
-    set(CMAKE_FIND_LIBRARY_PREFIXES ${_saved_lib_prefix}) # restore prefix
 
     if (${_library_uc}_LIBRARIES)
         set(${_component}_LIBRARY ${${_library_uc}_LIBRARIES}) # fake as if we call find_library
diff --git a/components/files/windowspath.cpp b/components/files/windowspath.cpp
index 2354e6f31..36290db9a 100644
--- a/components/files/windowspath.cpp
+++ b/components/files/windowspath.cpp
@@ -6,7 +6,7 @@
 
 #include <shlobj.h>
 #include <shlwapi.h>
-#include <WinReg.h>
+#include <winreg.h>
 
 #include <boost/locale.hpp>
 namespace bconv = boost::locale::conv;
